# Unref the dbus proxy when finalizing the session object, so that they are not leaked
# Ubuntu: https://launchpad.net/bugs/284229
# Upstream: https://bugs.freedesktop.org/show_bug.cgi?id=18330
Index: consolekit-0.2.10/src/ck-session.c
===================================================================
--- consolekit-0.2.10.orig/src/ck-session.c	2008-02-12 21:28:57.000000000 +0100
+++ consolekit-0.2.10/src/ck-session.c	2009-03-29 21:20:26.000000000 +0200
@@ -1058,6 +1058,8 @@
 
         session_remove_activity_watch (session);
 
+        g_object_unref (session->priv->bus_proxy);
+
         g_free (session->priv->id);
         g_free (session->priv->cookie);
         g_free (session->priv->seat_id);
